Bmx.CommandLine/CommandLine.cs

```diff
@@ -5,14 +5,20 @@
 using System.CommandLine.Parsing;
 using System.Threading.Tasks;
 using Bmx.Core;
+using Bmx.Core.State;
 
 namespace Bmx.CommandLine {
-	public class CommandLine {
-		private readonly IBmxCore _bmx;
+	public class CommandLine<TAuthenticateState, TAuthenticatedState, TAccountState, TRoleState>
+		where TAuthenticateState : IAuthenticateState
```

Comment:
Is this syntax related to .NET Core DI? Never seen it before.

Reply:
This is a generic type constraint, I think it's also in .NET Framework.  Usually these `T` style arguments can be any time, Adipa is saying `TAuthenticateState` has to be an implementation of `IAuthenticateState`.

Reply:
Ah cool, thanks.